agreement/player.go

```diff
@@ -568,8 +572,12 @@ func (p *player) handleMessageEvent(r routerHandle, e messageEvent) (actions []a
 		}
 		up := e.Input.UnauthenticatedProposal
 
-		a := relayAction(e, protocol.ProposalPayloadTag, compoundMessage{Proposal: up, Vote: uv})
-		actions = append(actions, a)
+		// relay as the proposer
+		if e.TaskIndex == 1 {
```

Comment:
I'm sure there is a more elegant way to determine if this proposal was locally generated or not. One way to do so is to use the e.Input.MessageHandle and see if it's zero; messages that came from the "outside" would have this as a non-zero value. @derbear might have other suggestions.